src/lib/server/firebaseUtils.ts

```diff
@@ -29,7 +28,7 @@ type CheckSessionAuthOptions = {
  */
 export const checkSessionAuth = async (cookies: Cookies, options: CheckSessionAuthOptions = {}) => {
   const { loginRedirect, authFunction } = options;
-  const successRedirect = loginRedirect ? `/login?successRedirect=${encodeURIComponent(loginRedirect)}` : '/login';
+  const successRedirect = loginRedirect ? `/login?successRedirect=${encodeURIComponent(loginRedirect.replace('?', '&'))}` : '/login';
```

Comment:
here appending any existing search Params on the `loginRedirect` replacing `?` with `&` in the url.

let me know if you think there's a different way i should address this!